npe2/manifest/schema.py

```diff
@@ -147,14 +147,47 @@ def _check_engine_version(cls, values: dict) -> dict:
             )
         return values
 
+    @validator("display_name")
+    def validate_display_name(cls, v):
+
+        regex = r"^[^\W_][\w -~]{1,38}[^\W_]$"
+        if not bool(re.match(regex, v)):
+            raise ValueError(
+                f"{v} is not a valid display_name.  The display_name must "
+                "be 3-40 characters long, containing printable word characters, "
+                "and must not begin or end with an underscore, white space, or "
+                "non-word character."
+            )
+        return v
+
     @root_validator
     def _validate_root(cls, values: dict) -> dict:
         invalid_commands = []
+        invalid_python_names = []
         if values.get("contributions") is not None:
             for command in values["contributions"].commands or []:
-                if not command.id.startswith(values["name"]):
+                id_start_actual = command.id.split(".")[0]
+                if values["name"] != id_start_actual:
```

Comment:
```suggestion
                if not command.id.startswith(values["name"]):
```

Given what I suggested below, I like the way the predicate was originally done.

Reply:
The reason I had changed this was because leaving it with `startswith` would allow something with the name `napari-svg.get_writer` to allow something `napari-svg123.get_writer`.  If we don't care that something can trail before the `.`, then this will work great.

Reply:
good point! i take it back.